package/MDAnalysis/lib/distances.py

```diff
@@ -401,6 +433,11 @@ def self_distance_array(reference: Union[npt.NDArray, 'AtomGroup'],
              args=(reference, distances),
              backend=backend)
 
+    if backend == 'distopia':
+        # mda expects the result to be in float64, so we need to convert it back
+        # to float64, change for 3.0, see #3707
+        distances = distances.astype(np.float64)
```

Comment:
This may create a copy.

Reply:
You may have to instead put the numbers back into the original array to make in-place work.
```python
result[:] = distances

...

return result
```
or something of that kind.